contracts/main/price-feeders/DelayFathomOraclePriceFeed.sol

```diff
@@ -52,48 +50,88 @@ contract DelayFathomOraclePriceFeed is PausableUpgradeable, IFathomOraclePriceFe
     event LogSetPriceLife(address indexed _caller, uint256 _second);
     event LogSetTimeDelay(address indexed _caller, uint256 _second);
 
+    function readPrice() external view override returns (bytes32) {
+        return bytes32(delayedPrice);
+    }
+
+    function peekPrice() external override returns (bytes32, bool) {
+       return _peekPrice();
+    }
+
+    function isPriceOk() external view override returns (bool) {
+        return _isPriceOk();
+    }
+
+    function setAccessControlConfig(address _accessControlConfig) external onlyOwner {
+        IAccessControlConfig(_accessControlConfig).hasRole(IAccessControlConfig(_accessControlConfig).OWNER_ROLE(), msg.sender); // Sanity Check Call
+        accessControlConfig = IAccessControlConfig(_accessControlConfig);
+    }
+
     function setPriceLife(uint256 _second) external onlyOwner {
-        require(_second >= 5 minutes && _second <= 1 days, "FathomOraclePriceFeed/bad-price-life");
+        require(_second >= timeDelay && _second >= 5 minutes && _second <= 1 days, "FathomOraclePriceFeed/bad-price-life");
         priceLife = _second;
         emit LogSetPriceLife(msg.sender, _second);
     }
 
     function setTimeDelay(uint256 _second) external onlyOwner {
-        require(_second >= 5 minutes && _second <= 1 days, "FathomOraclePriceFeed/bad-delay-time");
+        require(_second <= priceLife &&_second >= 5 minutes && _second <= 1 days, "FathomOraclePriceFeed/bad-delay-time");
```

Comment:
2.3.3 Time delay can be higher than price life in DelayFathomOraclePriceFeed - fixed, by adding validation that checks if timeDelay we set in not bigger then price life